scipy/sparse/compressed.py

```diff
@@ -835,24 +835,36 @@ def _insert_many(self, i, j, x):
 
         self.check_format(full_check=False)
 
-    def _get_single_element(self,row,col):
+    def _get_single_element(self, row, col):
         M, N = self.shape
         if (row < 0):
             row += M
         if (col < 0):
             col += N
         if not (0 <= row < M) or not (0 <= col < N):
-            raise IndexError("index out of bounds")
+            raise IndexError("index out of bounds: 0<=%d<%d, 0<=%d<%d" %
+                             (row, M, col, N))
 
-        major_index, minor_index = self._swap((row,col))
-
-        # TODO make use of sorted indices (if present)
+        major_index, minor_index = self._swap((row, col))
 
         start = self.indptr[major_index]
-        end = self.indptr[major_index+1]
-        # can use np.add(..., where) from numpy 1.7
-        return np.compress(minor_index == self.indices[start:end],
-                           self.data[start:end]).sum(dtype=self.dtype)
+        end = self.indptr[major_index + 1]
+
+        if self.has_sorted_indices:
+            # Copies may be made, if dtypes of indices are not identical
+            minor_index = np.asarray(minor_index, dtype=self.indices.dtype)
+            insert_pos_left = np.searchsorted(self.indices[start:end],
+                                              minor_index, side='left')
+            insert_pos_right = np.searchsorted(self.indices[start:end],
+                                               minor_index, side='right')
```

Comment:
Good idea! I think we need to do the following, however:

`insert_pos_right = insert_pos_left + np.searchsorted(minor_indices[insert_pos_left:], minor_index, side='right')`

Otherwise, the indices do not match.